segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -779,115 +781,67 @@ public void readTableEntries(WireCommands.ReadTableEntries readTableEntries) {
-                                                          ArrayView value = e.getValue();
-                                                          val valueWireCommand = new WireCommands.TableValue(wrappedBuffer(value.array(), value.arrayOffset(),
-                                                                                                                           value.getLength()));
-                                                          return new AbstractMap.SimpleImmutableEntry<>(keyWireCommand, valueWireCommand);
-                                                      })
-                                                      .collect(toList());
-                      }
-
-                      connection.send(new WireCommands.TableEntriesRead(readTableEntries.getRequestId(), segment,
-                                                                        new WireCommands.TableEntries(wireCommandEntries),
-                                                                        continuationToken.get()));
-                      this.tableStatsRecorder.iterateEntries(readTableEntries.getSegment(), entries.size(), timer.getElapsed());
-                  }).exceptionally(e -> handleException(readTableEntries.getRequestId(), segment, operation, e));
-    }
-
-    private int getTableKeyBytes(String segment, Collection<TableKey> keys, int continuationTokenLength) {
-        int headerLength = WireCommands.TableKeysRead.GET_HEADER_BYTES.apply(keys.size());
-        int segmentLength = segment.getBytes().length;
-        int dataLength = keys.stream().mapToInt(value -> value.getKey().getLength() + Long.BYTES).sum();
-        return continuationTokenLength + headerLength + segmentLength + dataLength;
-    }
-
-    private int getTableEntryBytes(String segment, Collection<TableEntry> items, int continuationTokenLength) {
-        int headerLength = WireCommands.TableEntriesRead.GET_HEADER_BYTES.apply(items.size());
-        int segmentLength = segment.getBytes().length;
-        int dataLength = items.stream().mapToInt(value -> {
-            return value.getKey().getKey().getLength() // key
-                    + Long.BYTES // key version
-                    + value.getValue().getLength(); // value
-        }).sum();
-        return headerLength + segmentLength + dataLength + continuationTokenLength;
-    }
-
-    private ArrayView getArrayView(ByteBuf buf) {
-        final int length = buf.readableBytes();
-        if (buf.hasArray()) {
-            return new ByteArraySegment(buf.array(), buf.readerIndex(), length);
-        } else {
-            byte[] bytes;
-            bytes = new byte[length];
-            buf.getBytes(buf.readerIndex(), bytes);
-            return new ByteArraySegment(bytes, 0, length);
-        }
-    }
+                .thenCompose(itr -> itr.collectRemaining(
+                        e -> {
+                            if (entries.size() >= suggestedEntryCount || msgSize.get() >= MAX_READ_SIZE) {
```

Comment:
Same question here about the sync block.

Reply:
see above

Reply:
I will give the same answer since it's the same pattern used.